src/main/scala/state/Terms.scala

```diff
@@ -2198,6 +2198,17 @@ object Domain extends CondFlyweightTermFactory[(String, Term), Domain] {
   override def actualCreate(args: (String, Term)): Domain = new Domain(args._1, args._2)
 }
 
+class HasDomain(val field: String, val fvf: Term) extends Term with ConditionalFlyweight[(String, Term), HasDomain] {
+  utils.assertSort(fvf, "field value function", "FieldValueFunction", _.isInstanceOf[sorts.FieldValueFunction])
+
+  val sort = sorts.Bool
+  override val equalityDefiningMembers: (String, Term) = (field, fvf)
+}
+
+object HasDomain extends CondFlyweightTermFactory[(String, Term), HasDomain] {
+  override def actualCreate(args: (String, Term)): HasDomain = new HasDomain(args._1, args._2)
+}
```

Comment:
@marcoeilers can you please add, as a comment, an explanation of what this term means? (Your recent message to me would be a possible candidate)